client/v3/snapshot/v3_snapshot.go

```diff
@@ -64,6 +64,7 @@ func SaveWithVersion(ctx context.Context, lg *zap.Logger, cfg clientv3.Config, d
 	if err != nil {
 		return "", fmt.Errorf("could not open %s (%v)", partpath, err)
 	}
+	defer f.Close()
```

Comment:
it sounds like an easy fix, but how would the rename operation below work on an open file?

Reply:
Joining the errors should work :) 

Ref: `https://pkg.go.dev/errors#Join`